src/main/perl/ComponentProxyList.pm

```diff
@@ -146,31 +172,42 @@ sub run_all_components
             $self->set_state($name, "");
 
             my $ret = $comp->executeConfigure();
-            if (!defined($ret)) {
-                my $err = "cannot execute configure on component " . $name;
-                $self->error($err);
-                $status->{'ERRORS'}++;
-                $status->{ERR_COMPS}->{$name} = 1;
-                $self->set_state($name, $err);
-            } else {
+            if (defined($ret)) {
+                # errors before warnings (errors can be downgraded to warnings)
                 if ($ret->{'ERRORS'}) {
-                    $status->{ERR_COMPS}->{$name} = $ret->{ERRORS};
-                    $self->set_state($name, $ret->{ERRORS});
+                     if ($nodepsnoerrors) {
+                         # Convert errors in warnings
+                         # TODO do we set the state to failed?
+                         $self->warn("Errors from $name are downgraded to warnings (nodepsnoerrors is set). ",
+                                     "State is not cleared (as something went wrong), but also not set.");
```

Comment:
i'll reformutlate the emssages, but do we
a. set the state to fail (as something wnet wrong)
b. clear the state (as we ignore the error)
c. nothing (and in that case i propose we log the fact that we doon't change the state)
?